### PR TITLE
Closes #151: Show headings in the sidebar

### LINKED ISSUE #151: Website: Change the hash while scrolling
[Issuehunt badges](https://issuehunt.io/r/saberland/saber/issues/151)

Thanks to #145, we can also do this now:

屏幕快照 2019-05-07 下午11 43 55

- [ ] Show headings
- [ ] Detect scrolling
- [ ] Make sidebar section collapsible

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $50.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($50.00)
### Submitted pull Requests
- [#334 Fix #151: Change the hash while scrolling](https://issuehunt.io/r/saberland/saber/pull/334)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/saberland/saber/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by MarchWorks**
**Summary**
Closes #151 
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [x] Other, please describe:

If changing the UI/CSS related code, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
- [x] Show headings
- [x] Change the hash depending on the scroll position
- [x] Make sidebar sections's collapsible

### CODE REVIEW

**Comment by krmax44** [comment]
> Make sidebar section collapsible (???? sidebar section is already collapsible when the screen size is small)

I think it means adding a hamburger menu style button to allow collapsing it even on a desktop

**Comment by MarchWorks** [comment]
I just think that there is no need to do that because it became collapsible when the window inner width is inferior to 768px

**Comment by egoist** [comment]
屏幕快照 2019-05-09 下午9 54 35

I mean something like this.

**Comment by MarchWorks** [comment]
Sorry I am not a fit pro. How to get ride of `No staged files match any of provided globs` so that the ckecks pass

**Comment by MarchWorks** [comment]
Seems that webpack can't import `./somthing.vue` that's what I understood from the last commit

**Comment by MarchWorks** [comment]
I am thinking of changing the hash depanding on the nearest anchore to the middle of the screen. What do you say?

File: website/src/components/Sidebar.vue [hunk 1]
```diff
@@ -2,12 +2,18 @@
   <nav class="sidebar">
     <div class="items">
       <div class="item" v-for="(item, i) in items" :key="i">
-        <div class="item-title">{{ item.title }}</div>
-        <div class="item-children">
+        <div class="item-wrapper">
+          <div class="item-title">{{ item.title }}</div>
+          <button v-on:click="expandCollapse(i)" :class="'expand' + (itms.includes(i) ? ' collapse' : '')">
```
Comment (krmax44) [inline website/src/components/Sidebar.vue]:
```suggestion
          <button @click="expandCollapse(i)" :class="{ expand: true, collapse: itms.includes(i) }">
```

File: website/src/components/Sidebar.vue [hunk 2]
```diff
@@ -16,32 +22,38 @@
...
-        }
-      }
-      return res
+  components: {
+    SubLinks
+  },
+
+  data() {
+    return {
+      itms: []
```
Comment (krmax44) [inline website/src/components/Sidebar.vue]: That's too confusing. `items` and `itms` should be easily distinguishable.

File: website/src/components/Sidebar.vue [hunk 3]
```diff
@@ -16,32 +22,38 @@
...
     }
   },
 
+
   methods: {
     isActive(link) {
```
Comment (krmax44) [inline website/src/components/Sidebar.vue]: should be a computed property

File: website/src/layouts/docs.vue [hunk 4]
```diff
@@ -45,6 +45,26 @@ export default {
   },
 
   computed: {
+  	markdownHeadings() {
+      var arr = [...this.page.markdownHeadings]
```
Comment (krmax44) [inline website/src/layouts/docs.vue]: use `const`/`let`

File: website/src/components/Sidebar.vue [hunk 5]
```diff
@@ -77,13 +89,55 @@ export default {
...
+.expand {
+  width: 35px;
+  border: 0;
+  outline: none;
+  background: transparent;
+  cursor: pointer;
+  & > span {
+    position: relative;
+    display: block;
+    &:before, &:after {
```
Comment (krmax44) [inline website/src/components/Sidebar.vue]: line breaks should be inserted before `& ...`

Comment (MarchWorks) [reply website/src/components/Sidebar.vue]: ?

File: website/src/components/Sidebar.vue (hunk 3 above)
Comment (MarchWorks) [reply website/src/components/Sidebar.vue]: doesn't work when I change it to a computed property

**Comment by MarchWorks** [comment]
> when scrolling, the page jumps

`history.pushState(null, null, hash)` will do the job with some other changes

**Comment by MarchWorks** [comment]
check the docs page

**Pull request closed without merging**